code/modules/client/preference_setup/general/01_basic.dm

```diff
@@ -167,4 +217,33 @@
 			pref.metadata = sanitize(new_metadata)
 			return TOPIC_REFRESH
 
+	else if(href_list["ipc_tag"])
+		var/choice = alert(user, "Do you wish for your IPC to have a tag?\n\nWARNING: Being an untagged IPC in Tau space is highly illegal!", "IPC Tag", "Yes", "No")
+		if(CanUseTopic(user))
+			if(choice == "Yes")
+				pref.machine_tag_status = TRUE
+			else
+				pref.machine_tag_status = FALSE
+			return TOPIC_REFRESH
+
+	else if(href_list["serial_number"])
+		var/new_serial_number = sanitize(input(user, "Enter what you want to set your serial number to.", "IPC Serial Number", pref.machine_serial_number) as message|null)
+		new_serial_number = uppertext(dd_limittext(new_serial_number, 12))
+		if(new_serial_number && CanUseTopic(user))
+			pref.machine_serial_number = sanitize(new_serial_number)
+			return TOPIC_REFRESH
+
+	else if(href_list["generate_serial"])
+		if(pref.real_name)
+			var/generated_serial = uppertext(dd_limittext(md5(pref.real_name), 12))
+			pref.machine_serial_number = generated_serial
+			return TOPIC_REFRESH
+
+	else if(href_list["ownership_status"])
+		var/static/list/ownership_options = list(IPC_OWNERSHIP_COMPANY, IPC_OWNERSHIP_PRIVATE, IPC_OWNERSHIP_SELF)
+		var/new_ownership_status = input(user, "Choose your IPC's ownership status.", "IPC Ownership Status") as null|anything in ownership_options
+		if(new_ownership_status && CanUseTopic(user))
+			pref.machine_ownership_status = new_ownership_status
+			return TOPIC_REFRESH
```

Comment:
You need to check there aswell if the user is able to edit the tag information (as currently that would be possible via href hacking because you only validated the option to do so in the gui, but not the data sent from the gui)